egs/wsj/s5/steps/nnet3/train_dnn.py

```diff
@@ -160,6 +160,10 @@ def train(args, run_opts, background_process_handler):
     num_jobs = common_lib.get_number_of_jobs(args.ali_dir)
     feat_dim = common_lib.get_feat_dim(args.feat_dir)
     ivector_dim = common_lib.get_ivector_dim(args.online_ivector_dir)
+    num_cmn_offsets = common_lib.get_num_cmn_offsets(args.feat_dir)
+    if num_cmn_offsets > 0:
+      assert(ivector_dim % num_cmn_offsets == 0)
+      ivector_dim = ivector_dim / num_cmn_offsets
```

Comment:
once you make the changes I'm suggesting about how the iVectors are stored in scp files, this code change won't be necessary as you'd get the correct iVector dim from the 'get_ivector_dim' call.